libraries/Network/src/NetworkEvents.cpp

```diff
@@ -134,10 +134,73 @@ void NetworkEvents::checkForEvent() {
   free(event);
 }
 
+uint32_t NetworkEvents::findEvent(NetworkEventCb cbEvent, arduino_event_id_t event) {
+  uint32_t i;
+
+  if (!cbEvent) {
+    return cbEventList.size();
+  }
+
+  for (i = 0; i < cbEventList.size(); i++) {
+    NetworkEventCbList_t entry = cbEventList[i];
+    if (entry.cb == cbEvent && entry.event == event) {
+      break;
+    }
+  }
+  return i;
+}
+
+template<typename T, typename... U> static size_t getStdFunctionAddress(std::function<T(U...)> f) {
+  typedef T(fnType)(U...);
+  fnType **fnPointer = f.template target<fnType *>();
+  if (fnPointer != nullptr) {
+    return (size_t)*fnPointer;
+  }
+  return (size_t)fnPointer;
+}
+
+uint32_t NetworkEvents::findEvent(NetworkEventFuncCb cbEvent, arduino_event_id_t event) {
+  uint32_t i;
+
+  if (!cbEvent) {
+    return cbEventList.size();
+  }
+
+  for (i = 0; i < cbEventList.size(); i++) {
+    NetworkEventCbList_t entry = cbEventList[i];
+    if (getStdFunctionAddress(entry.fcb) == getStdFunctionAddress(cbEvent) && entry.event == event) {
```

Comment:
Why not place the result of `getStdFunctionAddress(cbEvent)` outside the for loop?

Reply:
wasn't before, it's OK not to change that now also